tiatoolbox/tools/patchextraction.py

```diff
@@ -0,0 +1,361 @@
+    #     )
+    #
+    #     num_patches_img = len(labels) * self.num_examples_per_patch
+    #     img_patches = np.zeros(
+    #         (num_patches_img, patch_h, patch_w, image.shape[2]), dtype=image.dtype
+    #     )
+    #     labels = [None] * num_patches_img
+    #     class_id = [None] * num_patches_img
+    #
+    #     cell_tot = 1
+    #     iter_tot = 0
+    #     for row in labels:
+    #         patch_label = row[0]
+    #         cell_location = np.array([row[2], row[1]], dtype=np.int)
+    #         cell_location[0] = (
+    #             cell_location[0] + self.pad_y - 1
+    #         )  # Python index starts from 0
+    #         cell_location[1] = (
+    #             cell_location[1] + self.pad_x - 1
+    #         )  # Python index starts from 0
+    #         if self.num_examples_per_patch > 1:
+    #             root_num_examples = np.sqrt(self.num_examples_per_patch)
+    #             start_location = -int(root_num_examples / 2)
+    #             end_location = int(root_num_examples + start_location)
+    #         else:
+    #             start_location = 0
+    #             end_location = 1
+    #
+    #         for h in range(start_location, end_location):
+    #             for w in range(start_location, end_location):
+    #                 start_h = cell_location[0] - h - int((patch_h - 1) / 2)
+    #                 start_w = cell_location[1] - w - int((patch_w - 1) / 2)
+    #                 end_h = start_h + patch_h
+    #                 end_w = start_w + patch_w
+    #                 labels[iter_tot] = patch_label
+    #                 class_id[iter_tot] = cell_tot
+    #              img_patches[iter_tot, :, :, :] = image[start_h:end_h, start_w:end_w]
+    #                 iter_tot += 1
+    #
+    #         cell_tot += 1
+    #     return img_patches, labels, class_id
+
+    def merge_patches(self, patches=None):
+        """Merge patch is not supported by :obj:`PointsPatchExtractor`.
+        Calling this function for :obj:`PointsPatchExtractor` will raise an error. This
+        overrides the merge_patches function in the base class :obj:`PatchExtractor`
+
+        """
+        raise MethodNotSupported(
+            message="Merge patches not supported for PointsPatchExtractor"
+        )
+
+
+def convert_input_image_for_patch_extraction(input_image):
+    """Convert PatchExtraction input_image attribute to ndarray or :obj:`WSIReader`.
+
+    Args:
+        input_image(str, pathlib.Path, ndarray): input image for patch extraction.
+
+    Return:
+        ndarray, WSIReader: a numpy arrary or an object of type :obj:`WSIReader`.
+
+    """
+    if not isinstance(input_image, np.ndarray):
+
+        if isinstance(input_image, pathlib.Path):
+            input_image = str(input_image)
+
+        if isinstance(input_image, str):
+            _, _, suffix = split_path_name_ext(input_image)
+
+            if suffix in (".jpg", ".png"):
+                input_image = imread(input_image)
```

Comment:
Wrapper class

Reply:
Added in this commit https://github.com/TIA-Lab/tiatoolbox/commit/ae7bb3079412829b3b8a79c8126a202c4170c446